File: 0225-implement-stack-using-queues/0225-implement-stack-using-queues.py
class MyStack(object):

    def __init__(self):
               
        self.arr= []
        

    def push(self, x):
        
        self.arr.append(x)
        
            

        """
        :type x: int
        :rtype: None
        """
        

    def pop(self):
        
        return self.arr.pop(-1)
        """
        :rtype: int
        """
        

    def top(self):
        return self.arr[-1]
        """
        :rtype: int
        """
        

    def empty(self):
        
        return(len(self.arr) == 0)
        """
        :rtype: bool
        """
        


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()

File: 0225-implement-stack-using-queues/0225-implement-stack-using-queues.js
var Node = function(value) {
    this.value = value;
    this.next = null;
    this.prev = null;
};


var MyStack = function() {
    this.first = null;
    this.last = null;
    this.length = 0;
};

/** 
 * @param {number} x
 * @return {void}
 */
MyStack.prototype.push = function(x) {
    newNode = new Node(x);
    newNode.value = x;
    if(this.length === 0){
        this.first = newNode;
        this.last = this.first;
    } else{
        this.first.next = newNode;
        newNode.prev = this.first;
        this.first = this.first.next;
    }
    this.length += 1;
};

/**
 * @return {number}
 */
MyStack.prototype.pop = function() {
    let returnMe = this.first.value;
    if(this.length > 1){
        this.first = this.first.prev
        this.length -=1 ;
    }else if(this.length === 1){
        this.first = null;
        this.first = this.last;
        this.length -= 1;
    }
    return returnMe;
};

/**
 * @return {number}
 */
MyStack.prototype.top = function() {
    return this.first.value;
};

/**
 * @return {boolean}
 */
MyStack.prototype.empty = function() {
    return this.length === 0;
};

/** 
 * Your MyStack object will be instantiated and called as such:
 * var obj = new MyStack()
 * obj.push(x)
 * var param_2 = obj.pop()
 * var param_3 = obj.top()
 * var param_4 = obj.empty()
 */